src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -113,6 +123,34 @@ public override void ExecuteCmdlet()
         {
             var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.Profile);
 
+            if (this.ParameterSetName.Equals(MetadataParameterSet, StringComparison.Ordinal))
+            {
+                MetadataResponse metadataEndpoints = null;
+                ResourceManagerEndpoint = ARMEndpoint;
+                try
+                {
+                    metadataEndpoints = EnvironmentHelper.RetrieveMetaDataEndpoints(ResourceManagerEndpoint).Result;
+                    string domain = EnvironmentHelper.RetrieveDomain(metadataEndpoints.PortalEndpoint);
+                    ActiveDirectoryEndpoint = metadataEndpoints.authentication.LoginEndpoint.TrimEnd('/') + '/';
+                    ActiveDirectoryServiceEndpointResourceId = metadataEndpoints.authentication.Audiences[0];
+                    GalleryEndpoint = metadataEndpoints.GalleryEndpoint;
+                    GraphEndpoint = metadataEndpoints.GraphEndpoint;
+                    AzureKeyVaultDnsSuffix = string.Format("vault.{0}", domain).ToLowerInvariant();
+                    AzureKeyVaultServiceEndpointResourceId = string.Format("https://vault.{0}", domain).ToLowerInvariant();
+                    StorageEndpoint = domain;
+                    EnableAdfsAuthentication = metadataEndpoints.authentication.LoginEndpoint.TrimEnd('/').EndsWith("/adfs", System.StringComparison.OrdinalIgnoreCase);
+                }
+                catch (Exception ex)
+                {
+                    if (ex.InnerException != null)
```

Comment:
It throws aggregate exception if I don't unwrap this. The ArgumentException or CloudException thrown from RetrieveMetaDataEndpoints function gets converted  into AggregateException. In PowerShell, AggregateException did not give much info about the exception (any error message). Should we still throw as it is?